styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java

```diff
@@ -727,26 +726,9 @@ private void processPodUpdate(String podName, WorkflowInstance instance) {
       emitPodEvents(pod, runState.get());
     }
 
-    private void reconnect() {
-      LOG.warn("Re-establishing pod watcher");
-
-      try {
-        watch = client.pods()
-            .watch(this);
-      } catch (Throwable e) {
-        LOG.warn("Retry threw", e);
-        scheduleReconnect();
-      }
-    }
-
-    private void scheduleReconnect() {
-      scheduledExecutor.schedule(this::reconnect, RECONNECT_DELAY_SECONDS, TimeUnit.SECONDS);
-    }
-
     @Override
     public void onClose(KubernetesClientException e) {
       LOG.warn("Watch closed", e);
```

Comment:
In case of `HTTP_GONE`, fabric8 will issue a forced close and we will never get reconnected. :(

```
if (status.getCode() == HTTP_GONE) {
  webSocketRef.set(null);
  closeEvent(new KubernetesClientException(status));
  close(); // here everything is shutdown
  return;
}

@Override
  public void close() {
    logger.debug("Force closing the watch {}", this);
    closeEvent(null);
    closeWebSocket(webSocketRef.getAndSet(null));
    if (!executor.isShutdown()) {
      try {
        executor.shutdown();
        if (!executor.awaitTermination(1, TimeUnit.SECONDS)) {
          logger.warn("Executor didn't terminate in time after shutdown in close(), killing it in: {}", this);
          executor.shutdownNow();
        }
      } catch (Throwable t) {
        throw KubernetesClientException.launderThrowable(t);
      }
    }
  }
```

Nothing critical since we have a fallback polling. I guess we can reconnect ourselves in case of `HTTP_GONE` (this happened when upgrading GKE master node).